src/periphery/SablierMerkleFactory.sol

```diff
@@ -39,10 +41,41 @@ contract SablierMerkleLockupFactory is ISablierMerkleLockupFactory {
                          USER-FACING NON-CONSTANT FUNCTIONS
     //////////////////////////////////////////////////////////////////////////*/
 
-    /// @notice inheritdoc ISablierMerkleLockupFactory
+    /// @notice inheritdoc ISablierMerkleFactory
+    function createMerkleInstant(
+        MerkleBase.ConstructorParams memory baseParams,
+        uint256 aggregateAmount,
+        uint256 recipientCount
+    )
+        external
+        returns (ISablierMerkleInstant merkleInstant)
+    {
+        // Hash the parameters to generate a salt.
+        bytes32 salt = keccak256(
+            abi.encodePacked(
+                msg.sender,
+                baseParams.asset,
```

Comment:
i can't remember, was there a reason for not:
```solidity
bytes32 salt = keccak256(abi.encodePacked(msg.sender, abi.encode(baseParams)));
```

Reply:
It's because `baseParams.name` is stored in the form of `bytes32`. And the objective is to hash with the stored variables.

```solidity
abi.encodePacked(
    msg.sender,
    baseParams.asset,
    baseParams.expiration,
    baseParams.initialAdmin,
    abi.encode(baseParams.ipfsCID),
    baseParams.merkleRoot,
    bytes32(abi.encodePacked(baseParams.name))
)
```